lib/stages/RingMapMaker.cpp

```diff
@@ -471,17 +522,19 @@ void RedundantStack::main_thread() {
 
         // Check dataset id hasn't changed
         if (input_frame.dataset_id != input_dset_id) {
-            WARN("Input dataset ID has changed. Regenerating stack specs.");
             input_dset_id = input_frame.dataset_id;
             change_dataset_state(input_dset_id);
         }
 
-        const auto& stack_rmap = new_stack_state_ptr->get_rstack_map();
-        const auto& old_stack_map = old_stack_state_ptr->get_stack_map();
-        auto num_stack = new_stack_state_ptr->get_num_stack();
+        // Get the corresponding stack state
```

Comment:
In the if statement above, it's probably best to change the `input_frame.dataset_id != input_dset_id` clause into one like `dset_id_map.count(input_frame.dataset_id) == 0`

Reply:
right, thanks for catching this!